sdk/go/x/auto/local_workspace.go

```diff
@@ -972,3 +965,59 @@ func getStackSettingsName(stackName string) string {
 }
 
 const pulumiHomeEnv = "PULUMI_HOME"
+
+func readProjectSettingsFromDir(ctx context.Context, workDir string) (*workspace.Project, error) {
+	for _, ext := range settingsExtensions {
+		projectPath := filepath.Join(workDir, fmt.Sprintf("Pulumi%s", ext))
+		if _, err := os.Stat(projectPath); err == nil {
+			proj, err := workspace.LoadProject(projectPath)
+			if err != nil {
+				return nil, errors.Wrap(err, "found project settings, but failed to load")
+			}
+			return proj, nil
+		}
+	}
+	return nil, errors.New("unable to find project settings in workspace")
+}
+
+func getProjectSettings(
+	ctx context.Context,
+	projectName string,
+	opts []LocalWorkspaceOption,
+) (*workspace.Project, error) {
+	var optsBag localWorkspaceOptions
+	for _, opt := range opts {
+		opt.applyLocalWorkspaceOption(&optsBag)
+	}
+
+	// If the Project is included in the opts, just use that.
+	if optsBag.Project != nil {
+		return optsBag.Project, nil
+	}
+
+	// If WorkDir is specified, try to read any existing project settings before resorting to
+	// creating a default project.
+	if optsBag.WorkDir != "" {
+		proj, err := readProjectSettingsFromDir(ctx, optsBag.WorkDir)
+		if err == nil {
+			return proj, nil
```

Comment:
Minor, but I don't think we should round trip the YAML in the noop case as it can cause comments to be stripped.

If the user specifies a workDir with a pulumi.yaml, we will read in the contents and then write them back out onto disk when we create the workspace.

Reply:
Looks like this applies nodejs and elsewhere. We likely don't want to specify a projectSettings in this case, just avoid setting the default.

Reply:
I'm gonna merge this and make another PR with those changes.